api/src/config/index.ts

```diff
@@ -42,39 +42,29 @@ if (env === 'development') {
 // overrides are always as defined
 nconf.overrides({
   environment: env,
-  host: process.env.HOST || '127.0.0.1',
-  port: process.env.PORT || defaultPort,
-  minio: {
-    host: process.env.MINIO_HOST,
-    accessKey: process.env.MINIO_ACCESS_KEY,
-    secretKey: process.env.MINIO_SECRET_KEY,
-  },
+  port: process.env.PORT,
   db: {
-    user: process.env.POSTGRESQL_USER,
-    password: process.env.POSTGRESQL_PASSWORD,
-    host: process.env.POSTGRESQL_HOST,
+    host: process.env.POSTGRES_HOST,
+    port: process.env.POSTGRES_PORT,
+    database: process.env.APP_DB_NAME,
+    user: process.env.APP_DB_USER,
+    password: process.env.APP_DB_PASSWORD,
   },
   sso: {
     clientSecret: process.env.SSO_CLIENT_SECRET,
   },
-  agent: {
-    hostUrl: process.env.AGENT_URL,
-  },
-  session: {
-    key: process.env.SESSION_SECRET,
-  },
 });
 
 // load other properties from file.
 nconf
   .argv()
   .env()
-  .file({ file: path.join(__dirname, `${env}.json`) });
+  .file({ file: path.join(__dirname, `${fileName}`) });
```

Comment:
does this config file live relative to this directory? Is there a need for the config to be different in different environments (production, test etc)? If so, does this allow a configmap to be mounted

Reply:
I mounted it as a configmap now. This way we don't need to do a new build to change config params.